shell/browser/notifications/platform_notification_service.cc

```diff
@@ -116,10 +214,54 @@ void PlatformNotificationService::DisplayPersistentNotification(
     const GURL& service_worker_scope,
     const GURL& origin,
     const blink::PlatformNotificationData& notification_data,
-    const blink::NotificationResources& notification_resources) {}
+    const blink::NotificationResources& notification_resources) {
+  // SAP - 14036
+  absl::optional<int> proc_id = browser_client_->GetRenderFrameProcessID();
+  if (!proc_id.has_value())
+    return;
+
+  auto* presenter = browser_client_->GetNotificationPresenter();
+  if (!presenter)
+    return;
+
+  const size_t prev_notif_count = presenter->notifications().size();
+  const bool try_pending_deletion(notification_data.renotify &&
+                                  !notification_data.require_interaction);
+  presenter->CloseNotificationWithId(notification_id, try_pending_deletion);
+  const size_t curr_notif_count = presenter->notifications().size();
+
+  auto* delegate = new NotificationDelegateImpl(notification_id, origin);
+
+  auto notification = presenter->CreateNotification(delegate, notification_id);
+  if (notification) {
+    content::WebContents* web_ctx =
```

Comment:
Please take into according that non-persistent notifications are rather difficult as persistent

```
//non-persistent
void PlatformNotificationService::DisplayNotification(
    content::RenderFrameHost* render_frame_host,
```

```
//pesistent
void PlatformNotificationService::DisplayPersistentNotification(
    const std::string& notification_id,
    const GURL& service_worker_scope,
    const GURL& origin,```
    
 So we does not have content::RenderFrameHost* render_frame_host, for persistent notifications, only one way to find necessary content::WebContents is approach given into  DisplayPersistentNotification
 Be sure - this method browser_client_->GetRenderFrameProcessID(); returns only necessary process (I would propose to follow all call stack to understand this idea).
 
ADD: Also I can say more - this thing content::RenderFrameHost* render_frame_host, (I saw some patch to add this pointer) can be replaced on approach with finding  RenderFrameProcessID and than necessary WebContents, i.e we can exclude adding  content::RenderFrameHost* render_frame_host, at all and have native not patched implementation.

Reply:
The association here is to the wrong `RenderFrameHost`.  The correct thing to do here is to refactor our approach to creating `PlatformNotificationService`.

Currently the `PlatformNotificationService` class is a singleton, we should instead create one `PlatformNotificationService` per `BrowserContext` inside `ElectronBrowserContext::GetPlatformNotificationService`.  This allows the notification service to store a reference to the browser context that _constructed it_.  Then we use that to dispatch permission requests and make the breaking change to not provide `webContents` with those `WebNotificationAllowed` checks (this is already allowed by the API, just pass `nullptr` instead of `WebContents*`.

Then we can remove all this `proc_id` code and logic that tries to infer `BrowserContext` from rfh or from proc_id

Reply:
Hello @MarshallOfSound in my vision passing nullptr will lead to call 
```
void ElectronBrowserClient::WebNotificationAllowed(
    content::WebContents* web_contents,
    base::OnceCallback<void(bool, bool)> callback) {
  if (!web_contents) {
    std::move(callback).Run(false, false);
    return;
  }
```
that in it's turn has predefined usage of false. I do not agree with such variant, in my vision it will broke existing functionality